### PR TITLE
fix: add containerPort declaration for webhook in helm chart

### PR DESCRIPTION
**Pull request by zevisert**
Adds a missing port declaration for the container port serving the webhook to the helm chart. Bumps the chart version for this fix. 

Notice that this port _is correctly declared_ in the kustomize patches, it's just the helm chart that didn't expose this. 

Here's the port being declared for the kustomize manifests that enable the webhook: 
https://github.com/kubeflow/spark-operator/blob/252eddd2fac294de0fff6dc9e14e23574221f960/manifest/spark-operator-with-webhook.yaml#L51-L52

But here, the helm chart only creates a port for metrics, with no condition to add a port when `.Values.webhook.enable` is true 

https://github.com/kubeflow/spark-operator/blob/252eddd2fac294de0fff6dc9e14e23574221f960/charts/spark-operator-chart/templates/deployment.yaml#L49-L53

Closes: #1962

### CODE REVIEW

**Comment by zevisert** [comment]
Closes #1962

File: charts/spark-operator-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -51,10 +51,16 @@ spec:
           {{- toYaml .Values.envFrom | nindent 10 }}
         securityContext:
           {{- toYaml .Values.securityContext | nindent 10 }}
-        {{- if .Values.metrics.enable }}
+        {{- if or .Values.metrics.enable .Values.metrics.enable }}
```
Comment (MasonLegere) [inline charts/spark-operator-chart/templates/deployment.yaml:54]:
```suggestion
        {{- if or .Values.metrics.enable .Values.webhook.enable }}
```

Comment (zevisert) [reply charts/spark-operator-chart/templates/deployment.yaml:54]: Whoa, yep hahah, thanks @MasonLegere!

**Comment by MasonLegere** [comment]
@zevisert Eek I broke the CLA by being attached to that commit. Will sign in a second

**Comment by zevisert** [comment]
Thanks, I'm not sure if anyone knows yet what will happen with the CLA now that Google has donated this project to kubeflow; for now it's part of the CI that came with the donation and everyone is still signing it.

I can rebase you off that suggestion and use my name if you'd prefer not to sign the CLA at this time.

**Comment by andreyvelich** [comment]
Hi Folks, sorry for the confusions around Google CLA.
We are in the transition period to turn off Google CLA and fully rely on DCO when we add that as required check for all of Kubeflow repos.
We will disable the Google CLA check for Spark Operator repo soon.

cc @kubeflow/kubeflow-steering-committee

File: charts/spark-operator-chart/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: spark-operator
 description: A Helm chart for Spark on Kubernetes operator
-version: 1.2.1
+version: 1.2.2
```
Comment (yuchaoran2011) [inline charts/spark-operator-chart/Chart.yaml:4]: Could you bump up the chart version to 1.2.3? Another PR earlier that yours just took 1.2.2

Comment (zevisert) [reply charts/spark-operator-chart/Chart.yaml]: Done, had to force push since I forgot the git --signoff again

**Review by yuchaoran2011 — APPROVED** [review]
/lgtm

**Comment by AndrewChubatiuk** [comment]
@yuchaoran2011 could you please drop tag `v1beta2-1.4.0-3.5.0` from repo and rerun pipeline of this commit? CI was just fixed, but no charts or images were published yet

**Comment by yuchaoran2011** [comment]
Yeah I noticed that. I don't have the admin permissions and hence am unable to delete an existing tag. That said, bumping up the chart version without changing the appVersion is common. What would that prevent the chart from being published?

**Comment by zevisert** [comment]
Is that something I caused here? I had a merge conflict with master over the chart version; I resolved it by keeping the appVersion from master (`v1beta2-1.4.0-3.5.0`) and the bumped chart version from here (`1.2.3`)

**Comment by AndrewChubatiuk** [comment]
@zevisert just wanted to retrigger a build and put a tag here, cause it's the latest merged commit

**Comment by AndrewChubatiuk** [comment]
@zevisert created MR to publish chart independently https://github.com/kubeflow/spark-operator/pull/1964, but bumped both image and chart versions, cause `v1beta2-1.4.0-3.5.0` doesn't exist in registry

**Pull request merged**